### DISCUSSION TITLE
Cart Attribute is null in Discounts API

### DISCUSSION THREAD

**Discussion by Nikolow** [Shopify/function-examples#265] [Feedback: Discounts APIs]
Hello,

a few days ago I started to play with the discounts API (product-discounts function). I have tried to read the cart attributes, but the query always returns null. I have also tested to see if any cart attributes are set in the shop (via liquid code) and at least 3 cart attributes are set. It seems that the query is returning always null.

The query:
```
query Input 
{
  cart 
  {
    attribute{ key, value }
  }
}
```

The result:
```
{
  "cart": {
    "attribute": null
  }
}
```

Cart Attributes (check) Liquid Code:
```
<script>
      console.log("Cart Attributes: ");
      {% for attribute in cart.attributes %}console.log("{{ attribute | first }}: {{ attribute | last }}");{% endfor %}
</script>
```

Cart Attributes (check) console output:
![image](https://github.com/Shopify/function-examples/assets/32817112/9088ddd5-989a-40f8-931b-76ae95836159)

Am I missing something or this is an issue with the API?

**Comment by adampetro** [comment]
The `attribute` field is for fetching a single attribute, and you can specify which attribute you would like to fetch with the `key` argument. For example:
```graphql
query Input {
  cart {
    attribute(key: "some-key") { value }
  }
}
```

**Reply by Nikolow** [reply]
You are absolutely right. 
Is there a way to retrieve more than 1 attribute? (for example all attributes)

**Reply by adampetro** [reply]
You cannot retrieve all attributes. You can retrieve multiple by [aliasing the selections](https://graphql.org/learn/queries/#aliases) like this:
```graphql
query Input {
  cart {
    attribute1: attribute(key: "key1") { value }
    attribute2: attribute(key: "key2") { value }
  }
}
```
